### DISCUSSION TITLE
Add "Raw Value" to Field Label Menu

### DISCUSSION THREAD

**Discussion by benhaynes** [directus/directus#3072] [Feature Requests]
When you click on a field label on the Item Detail page, you get a context menu with options. I propose adding a "Raw Value" option there, that allows users to view/edit/copy/paste values. Useful for duplicating complex fields, like Repeater JSON. Not sure if/how this would work for relational values... but we can start with non-relational.

**Comment by Nitwel** [comment]
At the moment I'm thinking of an prop eg. `showRaw` that when set to true displays the raw value. So interfaces like toggle can just igonre it and interfaces like repeater, datetime, slider and so on can have a way of displaying raw values.

**Reply by rijkvanzanten** [reply]
I personally think every field should have the "Modify raw" in the dropdown on the field name. That can pop up a dialog with a monospace textarea to modify the raw data

**Reply by Nitwel** [reply]
This would be a very easy but poweful way of doing it. 👍 
The question here is if we want to just have a textarea for all fields or or make the type relative eg. None for buttons but code for json?

**Reply by rijkvanzanten** [reply]
Lets start with a monospaced textarea for all interfaces. The point of this is that you edit the "raw value" directly, so wrapping it in type appropriate inputs (/ interfaces) kinda defeats that purpose IMO

**Reply by benhaynes** [reply]
Easy to start with all values in a monospaced textarea or code editor... later, we can discuss removing/changing certain types if needed.

**Reply by rijkvanzanten** [reply]
The only thing we'll have to do is some typecasting on the front-end. Eg we gotta make sure we convert `"false"` to `false` before sending it back up to the API. With a textarea, everything will be a string

**Reply by Nitwel** [reply]
Yep, thats why I thought we don't need that option on a boolean.

**Comment by Nitwel** [comment]
https://github.com/directus/next/pull/301
